keras_cv/models/stable_diffusion/ptp_utils.py

```diff
@@ -0,0 +1,283 @@
+    """
+    cross_attention_count = 0
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if (
+            not cross_attention_count % 2
+            and "cross_attention" in submodule_name
+        ):
+            submodule._name = f"{submodule_name}_attn1"
+            cross_attention_count += 1
+        elif cross_attention_count % 2 and "cross_attention" in submodule_name:
+            submodule._name = f"{submodule_name}_attn2"
+            cross_attention_count += 1
+
+
+def update_cross_attn_mode(
+    diff_model: tf.keras.Model, mode: str, attn_suffix: str = "attn"
+):
+    """Update the mode control variable.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+        mode (str): The mode parameter can take 5 values:
+                    - save: to save the attention map.
+                    - use_last: to use the original cross attention maps.
+                    - edit: to calculate the attention map with respect to the edited prompt.
+                    - injection: to use the edited prompt cross attention maps
+                    - unconditional: to perform the standard attention computations.
+        attn_suffix (str): Suffix used to find the attention layer, by default "attn".
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if (
+            "cross_attention" in submodule_name
+            and attn_suffix in submodule_name.split("_")[-1]
+        ):
+            submodule.cross_attn_mode.assign(mode)
+
+
+def update_attn_weights_usage(diff_model: tf.keras.Model, use: bool):
+    """Update the mode control variable.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+        use (bool): Whether to use the prompt weights.
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if (
+            "cross_attention" in submodule_name
+            and "attn2" in submodule_name.split("_")[-1]
+        ):
+            submodule.use_prompt_weights.assign(use)
+
+
+def add_attn_weights(diff_model: tf.keras.Model, prompt_weights: np.ndarray):
+    """Assign the attention weights to the diffusion model's corresponding tf.variable.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
+        prompt_weights (np.ndarray): Weights of the attention tokens.
+    """
+    for submodule in diff_model.submodules:
+        submodule_name = submodule.name
+        if (
+            "cross_attention" in submodule_name
+            and "attn2" in submodule_name.split("_")[-1]
+        ):
+            submodule.prompt_weights.assign(prompt_weights)
+
+
+def put_mask_dif_model(
+    diff_model: tf.keras.Model, mask: np.ndarray, indices: np.ndarray
+):
+    """Assign the diffusion model's tf.variables with the passed mask and indices.
+
+    Args:
+        diff_model (tf.keras.Model): The diffusion model.
```

Comment:
lets just keep the typings to the docstring descriptions to keep stype consistent across our implementation.

Reply:
Fixed!